deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/AutoPromoter.java

```diff
@@ -278,8 +269,14 @@ public void promoteBuild(EnvironBean currEnvBean, DeployBean currDeployBean, int
             currEnvBean.getEnv_name());
         if (result.getResult() == PromoteResult.ResultCode.PromoteBuild &&
             StringUtils.isNotEmpty(result.getPromotedBuild())) {
-            safePromote(null, result.getPromotedBuild(), Constants.BUILD_STAGE, currDeployBean,
-                currEnvBean);
+            TagBean buildTagBean = getBuildTag(result.getPromotedBuild());
```

Comment:
But it can be tricky here. Say we have
build1, bad
build2, good
build3, current
We still want to promote to build2 as we are running on build3.  But if we change the code here, it will not promote to build2 and the candidate is always build1. I think it'd be better to change the SQL select for candidates build.